### PR TITLE
IBX-1430: Fix REST API reference

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-1430](https://issues.ibexa.co/browse/IBX-1430)
| Versions      | 4.1, 4.0

Fix several indentation errors that hide some parts or create non-existing routes.

- `/calendar/calendar/event` → `/calendar/event`
- `/commerce/basket/{basketId}/{basketId}/line` → `/commerce/basket/{basketId}/line`
- `/content/assets/assets/images` → `/content/assets/images`
- `/content/locations/{path}/{locationId}` → `/content/locations/{path}` (the two forms were existing and are now merged; the first one is not necessary as the location ID is always part of the location path.)
- `/content/locations/{path}/children` and `/content/locations/{path}/urlaliases` were absent from HTML output.
- `/content/objects`'s POST request example was badly indented.
- `/content/objectstategroups/objectstates` → `/content/objectstategroups/{objectStateGroupId}/objectstates`
- `/content/types/{contentTypeId}/{fieldDefinitionId}` → `/content/types/{contentTypeId}/fielddefinitions/{fieldDefinitionId}`
- remove non-existing `/content/views/{identifier}` (`/views/{viewId}` equivalent is not implemented)
- `/product/catalog/attributes/{id}`'s request body example was absent from HTML output.
- `/product/catalog/attributes/translation/{languageCode}` → `/product/catalog/attributes/translation/{id}/{languageCode}`
- `/product/catalog/attribute_groups`'s 403 descriptions were incomplete or erroneous.
- `/user/users`:
  - Fix query parameters descriptions when using `HEAD` method.
  - Move `?roleId=/user/roles/1` example to /user/groups because, on a clean install, the anonymous role `1` is not assigned to the Anonymous User but to its Anonymous User Group, and, [`/user/users?roleId` only retrieves users with the given role directly assigned to them](https://github.com/ibexa/rest/blob/main/src/lib/Server/Controller/User.php#L527).

### CODE REVIEW

**Comment by adriendupuis** [comment]
I found the missing `/content/locations/{path}/urlaliases` documentation during #1591. There is certainly more.

TODO: Found other wrongly indented sub-resources.

File: docs/api/rest_api_reference/input/ez-content-location.raml [hunk 1]
```diff
@@ -84,58 +84,58 @@ get:
...
+                    description: OK - returns the list of URL aliases.
+                    body:
+                        application/vnd.ibexa.api.UrlAliasRefList+xml:
+                            type: UrlAliasRefList
+                            exammple: examples/content/locations/path/urlaliases/GET/UrlAliasRefList.xml.example
+                400:
+                    description: Error - The user has no permission to read URL aliases.
+                401:
+                    description: Error - The Location was not found.
     /{locationId}:
```
[the remark this answers is not in the document]
Comment (adriendupuis) [reply docs/api/rest_api_reference/input/ez-content-location.raml]: Merged in 7a8c178268c415742fc0fe9cad485d9623fe0acc

File: docs/api/rest_api_reference/input/ez-content-types.raml [hunk 2]
```diff
@@ -210,6 +214,9 @@ get:
                 404:
                     description: Error - The Content Type draft does not exist.
         /fieldDefinitions:
+            get:
```
[the remark this answers is not in the document]
Comment (adriendupuis) [reply docs/api/rest_api_reference/input/ez-content-types.raml]: Reverted in f26f9c548531e89b8ea2b66fc5e312513ea89e2d and kept for #1642

[the remark this answers is not in the document]
Comment (adriendupuis) [reply docs/api/rest_api_reference/input/ez-content-types.raml]: (Commit comment was wrong: this is not ez-objectstategroups.raml but ez-content-types.raml)

File: docs/api/rest_api_reference/input/ez-user-users.raml [hunk 3]
```diff
@@ -35,13 +35,13 @@ head:
     description: Verifies if there are Users matching given filter.
     queryParameters:
         roleId:
-            description: Lists Users assigned to the given Role (e.g. GET /user/users?roleId=/user/roles/1).
+            description: Check is there are Users directly assigned to the given Role.
```
Comment (DominikaK) [inline docs/api/rest_api_reference/input/ez-user-users.raml:38]:
```suggestion
            description: Checks if there are Users directly assigned to the given Role.
```

File: docs/api/rest_api_reference/input/ez-user-users.raml [hunk 4]
```diff
@@ -35,13 +35,13 @@ head:
...
         remoteId:
-            description: Retrieves the User for the given remote ID (e.g. GET /user/users?remoteId=55dd9713db75145f374bbd0b4f60ad29).
+            description: Checks if there is a User for the given remote ID (e.g. GET /user/users?remoteId=55dd9713db75145f374bbd0b4f60ad29).
         login:
-            description: Retrieves the User for the given login (e.g. GET /user/users?login=editor).
+            description: Checks if there is a User for the given login (e.g. GET /user/users?login=editor).
         email:
-            description: Lists Users with the given email (e.g. GET /user/users?email=editor@example.com).
+            description: Check if there is a User with the given email (e.g. GET /user/users?email=editor@example.com).
```
Comment (DominikaK) [inline docs/api/rest_api_reference/input/ez-user-users.raml:44]:
```suggestion
            description: Checks if there is a User with the given email (e.g. GET /user/users?email=editor@example.com).
```

**Review by DominikaK — APPROVED** [review]

File: docs/api/rest_api_reference/input/examples/content/objects/POST/ContentCreate.xml.example [hunk 5]
```diff
@@ -22,15 +22,15 @@
       <fieldDefinitionIdentifier>intro</fieldDefinitionIdentifier>
       <languageCode>eng-GB</languageCode>
       <fieldValue>
-    <value key="xml"><![CDATA[<section xmlns="http://ez.no/namespaces/ezpublish5/xhtml5/edit"><p>draft draft</p></section>]]></value>
-   </fieldValue>
+        <value key="xml"><![CDATA[<section xmlns="http://ibexa.co/namespaces/ezpublish5/xhtml5/edit"><p>draft draft</p></section>]]></value>
```
Comment (adriendupuis) [inline docs/api/rest_api_reference/input/examples/content/objects/POST/ContentCreate.xml.example:25]: https://github.com/ibexa/fieldtype-richtext/blob/4.0/src/bundle/Resources/config/fieldtype_services.yaml#L23

File: docs/api/rest_api_reference/input/ez-user-groups.raml [hunk 6]
```diff
@@ -3,7 +3,7 @@ get:
     description: Loads User Groups for either an an ID or a remote ID or a Role.
     queryParameters:
         roleId:
-            description: Lists User Groups assigned to the given Role.
+            description: Lists User Groups assigned to the given Role (e.g. GET /user/users?roleId=/user/roles/1).
```
Comment (adriendupuis) [inline docs/api/rest_api_reference/input/ez-user-groups.raml:6]: Should be `/user/groups?roleId=/user/roles/1` 💩

File: docs/api/rest_api_reference/input/ez-user-users.raml [hunk 7]
```diff
@@ -35,13 +35,13 @@ head:
     description: Verifies if there are Users matching given filter.
     queryParameters:
         roleId:
-            description: Lists Users assigned to the given Role (e.g. GET /user/users?roleId=/user/roles/1).
+            description: Checks if there are Users directly assigned to the given Role.
         remoteId:
-            description: Retrieves the User for the given remote ID (e.g. GET /user/users?remoteId=55dd9713db75145f374bbd0b4f60ad29).
+            description: Checks if there is a User for the given remote ID (e.g. GET /user/users?remoteId=55dd9713db75145f374bbd0b4f60ad29).
```
Comment (adriendupuis) [inline docs/api/rest_api_reference/input/ez-user-users.raml:40]: In the examples, it should be `HEAD` instead of `GET`

**Pull request merged**